src/components/NavBar/utils.ts

```diff
@@ -30,25 +36,38 @@ type LayersCategoriesType = LayersCategoryType[];
 type MenuItemsType = MenuItemType[];
 
 function formatLayersCategories(layersList: {
-  [key: string]: string[];
+  [key: string]: Array<LayerKey | TableKey>;
 }): LayersCategoriesType {
-  return map(layersList, (layerKeys, layersListKey) => ({
-    title: startCase(layersListKey),
-    layers: layerKeys
-      .map(key => LayerDefinitions[key])
-      .filter((val): val is LayerType => Boolean(val)),
-
-    tables: layerKeys
-      .map(key => TableDefinitions[key])
-      .filter(val => Boolean(val)),
-  }));
+  return map(layersList, (layerKeys, layersListKey) => {
+    return {
+      title: startCase(layersListKey),
+      layers: layerKeys.filter(isLayerKey).map(key => LayerDefinitions[key]),
+      tables: layerKeys.filter(isTableKey).map(key => TableDefinitions[key]),
+    };
+  });
 }
 
-export const menuList: MenuItemsType = chain(appJSON)
-  .get('categories')
-  .map((layersCategories, categoryKey) => ({
-    title: startCase(categoryKey),
-    icon: icons[categoryKey],
-    layersCategories: formatLayersCategories(layersCategories),
-  }))
-  .value();
+export const menuList: MenuItemsType = map(
+  appJSON.categories,
+  (layersCategories, categoryKey) => {
+    Object.values(layersCategories)
+      .flat()
+      .forEach(layerOrTableKey => {
+        if (!isLayerKey(layerOrTableKey) && !isTableKey(layerOrTableKey)) {
+          throw new Error(
+            `Key ${layerOrTableKey} isn't valid layer or table in category: ${categoryKey} - prism.json`,
+          );
+        }
+      });
+    return {
+      title: startCase(categoryKey),
+      icon: icons[categoryKey],
+      layersCategories: formatLayersCategories(
+        // safe to cast - runtime checked above.
```

Comment:
I think this would be better written using a user defined guard function, something like this:

```typescript
function checkLayersCategories(layersCategories: Record<string, any>): layersCategories is Record<string, Array<LayerKey | TableKey> {
  // use your existing check, return a boolean
}

if (checkLayersCategory(layersCategories)) {
  return { ... }
}
```

It more clearly defines the semantic intent of what you're doing.